client/packages/system/src/Vaccination/utils.ts

```diff
@@ -5,18 +5,35 @@
 import { VaccinationDraft } from './api';
 import { VaccinationCardItemFragment } from './api/operations.generated';
 
-export const isPreviousDoseGiven = (
+export const getPreviousDoseStatus = (
   row: VaccinationCardItemFragment,
   items: VaccinationCardItemFragment[] | undefined
 ) => {
   const vaccineCourseId = row.vaccineCourseId;
-  if (!items) return false;
+  if (!items) return undefined;
   const itemsForCourse = items.filter(
     item => item.vaccineCourseId === vaccineCourseId
   );
   const doseIndex = itemsForCourse.findIndex(dose => dose.id === row.id);
-  if (doseIndex === 0) return true;
-  return itemsForCourse[doseIndex - 1]?.given;
+  if (doseIndex === 0) return undefined;
+  return itemsForCourse[doseIndex - 1]?.status;
```

Comment:
If it returns `undefined`, it means there is no previous dose (i.e. it's the first). If it returns `null`, it means the previous dose hasn't been entered at all (neither Given nor Not Given), which means the current dose will be skipping a dose.

Reply:
Lets add this to the code as a comment, I'd prefer to have specific status returned I wonder how hard that would be...